src/main/java/org/openstreetmap/atlas/generator/AtlasGenerator.java

```diff
@@ -190,87 +180,60 @@ public void start(final CommandMap command)
-        logger.info("\n\n********** SAVED THE LINE SLICED SUB ATLAS **********\n");
+                .mapToPair(AtlasGeneratorHelper.subatlas(AtlasGeneratorHelper.subAtlasFilter,
+                        AtlasCutType.SILK_CUT))
+                .filter(tuple -> tuple._2() != null).cache();
+        saveAsHadoop(lineSlicedSubAtlasRDD, AtlasGeneratorJobGroup.LINE_SLICED_SUB, output);
 
-        // Slice the raw Atlas and filter any null atlases
+        // Relation slice the line sliced Atlas and filter any null atlases
         final JavaPairRDD<String, Atlas> fullySlicedRawAtlasShardsRDD = countryRawAtlasRDD
                 .mapToPair(AtlasGeneratorHelper.sliceRawAtlasRelations(broadcastBoundaries,
-                        broadcastSharding, lineSlicedSubAtlasPath, lineSlicedAtlasPath, atlasScheme,
-                        sparkContext))
-                .filter(tuple -> tuple._2() != null);
-
-        // Persist the RDD and save the intermediary state
-        final String fullySlicedRawAtlasPath = getAlternateSubFolderOutput(output,
-                FULLY_SLICED_ATLAS_FOLDER);
-        fullySlicedRawAtlasShardsRDD.cache();
-        this.getContext().setJobGroup("3", "Raw Atlas Relation Slicing");
-        fullySlicedRawAtlasShardsRDD.saveAsHadoopFile(fullySlicedRawAtlasPath, Text.class,
-                Atlas.class, MultipleAtlasOutputFormat.class, new JobConf(configuration()));
-        logger.info("\n\n********** SAVED THE SLICED RAW ATLAS **********\n");
-
-        // Remove the partially sliced atlas RDD from cache since we've cached the sliced RDD
+                        broadcastSharding,
+                        getAlternateSubFolderOutput(output,
+                                AtlasGeneratorJobGroup.LINE_SLICED_SUB.getCacheFolder()),
+                        getAlternateSubFolderOutput(output,
+                                AtlasGeneratorJobGroup.LINE_SLICED.getCacheFolder()),
+                        atlasScheme, sparkContext))
+                .filter(tuple -> tuple._2() != null).cache();
+        saveAsHadoop(fullySlicedRawAtlasShardsRDD, AtlasGeneratorJobGroup.FULLY_SLICED, output);
+
+        // Remove the line sliced atlas RDD from cache since we've cached the fully sliced RDD
         lineSlicedAtlasRDD.unpersist();
         lineSlicedSubAtlasRDD.unpersist();
 
-        // Section the sliced raw Atlas
+        // Section the sliced Atlas
         final JavaPairRDD<String, Atlas> countryAtlasShardsRDD = fullySlicedRawAtlasShardsRDD
                 .mapToPair(AtlasGeneratorHelper.sectionRawAtlas(broadcastBoundaries,
                         broadcastSharding, sparkContext, broadcastLoadingOptions,
-                        fullySlicedRawAtlasPath, atlasScheme, tasks));
+                        getAlternateSubFolderOutput(output,
+                                AtlasGeneratorJobGroup.FULLY_SLICED.getCacheFolder()),
+                        atlasScheme, tasks))
+                .cache();
```

Comment:
Same here, have the cache as a separate statement?

Reply:
changed!